egs/wsj/s5/steps/cleanup/do_cleanup_segmentation.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# Copyright 2016  Vimal Manohar
+# Apache 2.0
+
+# This script demonstrates how to re-segment training data selecting only the 
+# "good" audio that matches the transcripts.
+# The basic idea is to decode with an existing in-domain acoustic model, and a
+# biased language model built from the reference, and then work out the
+# segmentation from a ctm like file.
+
+set -e 
+set -o pipefail 
+
+stage=0
+
+pad_length=5                  # Number of frames for padding the created 
+                              # subsegments
+max_silence_length=50         # Maxium number of silence frames above which they are removed and the segment is split
+max_incorrect_words=2         # Maximum number of incorrect words allowed in the segments
+min_correct_frames=0          # Minimum number of frames required in a subsegment to be kept
+max_segment_wer=20000         # Maximum WER% of an utterance, above which it is simply removed
+min_wer_for_splitting=-1      # Minimum WER% for a segment to be considered for splitting
+
+group_utt_words=              # If provided, build LMs on group of utterances 
+                              # that have at least these many words.  
+                              # Otherwise, build per-utterance LM
+
+# ngram options
+ngram_order=2                 
+interpolation_weights=0:0.1:0.9       # LM combination weights for 0-gram:1-gram:2-gram
```

Comment:
if you're using 0 weight on 0-gram, you could simplify this by just having a single variable unigram_interpolation_weight, and assign the rest to the bigram.